realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -191,7 +191,7 @@ private RealmQuery(BaseRealm realm, LinkView linkView, String className) {
      * @return {@code true} if still valid to use, {@code false} otherwise.
      */
     public boolean isValid() {
-        if (realm == null || realm.isClosed()) {
+        if (realm == null || realm.isClosed() /* this includes thread checking */) {
```

Comment:
I don't understand the comment `this includes thread checking`, `realm.isClosed` doesn't check for thread confinement.